themes/default/content/blog/going-beyond-with-advanced-infrastructure-as-code-use-cases/index.md

```diff
@@ -0,0 +1,122 @@
+---
+title: "Going Beyond With Advanced Infrastructure as Code Use Cases"
+date: 2023-11-21T03:53:33Z
+canonical_url: https://thenewstack.io/going-beyond-with-advanced-infrastructure-as-code-use-cases/
+draft: false
+meta_desc: "Explore advanced Infrastructure as Code (IaC) with Pulumi, covering cloud development, testing, policy enforcement, and the Automation API in a concise guide."
+meta_image: meta.png
+authors:
+    - joe-duffy
+tags:
+    - infrastructure-as-code
+    - fundamentals
+
+---
+
+*This is the third of a three-part series originally published on The New Stack. Read [Part 1](/blog/infrastructure-as-code-in-any-programming-language/) and [Part 2](/blog/walkthrough-of-adopting-infrastructure-as-code/).*
+
+Engineers who modernize their [Infrastructure as Code](/what-is/what-is-infrastructure-as-code/) with Pulumi get two classes of benefits:
+
+1. Infrastructure as Code to develop cloud infrastructure with code.
+2. Pulumi Cloud, which tames cloud infrastructure management at [scale](https://roadmap.sh/videos/scaling-the-unscalable).
+
+We’ve covered a fair bit of the first above, but have yet to scratch the surface for the second.
+
+Many Infrastructure as Code solutions require that you explicitly manage so-called “state.” This state is an artifact produced that keeps track of what you think your infrastructure looks like, as defined by your Infrastructure as Code, so that it can be easily compared to what your actual infrastructure looks like. This is how diffs and updates can be done. Every Infrastructure as Code tool stores this infrastructure state, which is really just metadata about all the cloud resources, properties and dependencies. However, how much the tool exposes you to it varies.
```

Comment:
```suggestion
Many Infrastructure as Code solutions require that you explicitly manage something typically referred to as “state.” This state is an artifact produced that keeps track of what you think your infrastructure looks like, as defined by your Infrastructure as Code, so that it can be easily compared to what your actual infrastructure looks like. This is how diffs and updates can be done. Every Infrastructure as Code tool stores this infrastructure state, which is really just metadata about all the cloud resources, properties and dependencies. However, how much the tool exposes you to it varies.
```